source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -194,6 +195,20 @@ SplitRequestPtr EvalRequest::create(Router& router, Common::Redis::RespValuePtr&
 
   const auto route = router.upstreamPool(incoming_request->asArray()[3].asString());
   if (route) {
+    int key_count;
+    try {
+      key_count = std::stoi(incoming_request->asArray()[2].asString());
```

Comment:
Please us `absl::SimpleAtoi` here. If we are going to handle int parse failure, please add a specific test for this.